test/e2e/storage/testsuites/base.go

```diff
@@ -172,8 +174,21 @@ func skipUnsupportedTest(driver TestDriver, pattern testpatterns.TestPattern) {
 	if !dInfo.SupportedFsType.Has(pattern.FsType) {
 		e2eskipper.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.FsType)
 	}
-	if pattern.FsType == "xfs" && framework.NodeOSDistroIs("gci", "cos", "windows") {
-		e2eskipper.Skipf("Distro doesn't support xfs -- skipping")
+	if pattern.FsType == "xfs" {
+		if framework.TestContext.Provider == "gke" && framework.NodeOSDistroIs("cos") {
+			// GKE can support XFS with COS node images, sometimes.
+			major, minor, err := getKubeServerVersion()
+			if err != nil {
+				klog.Fatalf("%v", err)
+			}
+			if major == 1 && minor < 18 {
```

Comment:
In normal GKE CI, we align minor versions of k8s and e2e.test. So I'm not sure we need the version check. For pdcsi tests, we can toggle xfs in our own test runner.

Reply:
If I understand this test skip code correctly, it will disable XFS no matter what the driver setup is. So I think we have to change this logic here. Or add plumbing for a driver to tell the e2e test that XFS is okay no matter the node image type.

I'm not sure what you mean by the version alignment---do you mean that as long as we don't back port this past 1.18, we don't have to check the version at all and could just enable it unconditionally for COS on GKE? That would be cool & simpler.

Reply:
yes I mean we can remove the check entirely except for windows maybe.

For pdcsi, we can set supported fs types in the driverconfig.

Reply:
Yeah but since we use head of e2e.test in pdcsi, we may need to add logic to not run xfs tests on versions < 1.18